adal/src/main/java/com/microsoft/aad/adal/AuthenticationRequestState.java

```diff
@@ -31,12 +31,15 @@
     private boolean mCancelled = false;
 
     private AuthenticationRequest mRequest = null;
-    
+
+    private final APIEvent mAPIEvent;
+
     public AuthenticationRequestState(int requestCallbackId, AuthenticationRequest request,
-            AuthenticationCallback<AuthenticationResult> delegate) {
+            AuthenticationCallback<AuthenticationResult> delegate, final APIEvent apiEvent) {
```

Comment:
repeat which part? having a map in the authenticationContext, and retrieve the the api event by the request id won't introduce repeat code I think. 

---
In reply to: [76166000](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/703#discussion_r76166000) [](ancestors = 76166000,76135343)